[BUG] - Scheduling a job and calling get current job before every/cron panics

### Describe the bug

<!---A clear and concise description of what the bug is.--->

When scheduling a job, if I call a func that calls `getCurrentJob()` prior to the `newJob()` func being called, a slice index panic occurs.

### To Reproduce

Steps to reproduce the behavior:

```go
s := gocron.NewScheduler(time.UTC)
s.WaitForSchedule().Every("1m").Do(func() {})
```

output:
```
panic: runtime error: index out of range [-1]
```

### Version

<!---The release version or commit SHA you're using.--->

v1.7.1

### Expected behavior

<!---A clear and concise description of what you expected to happen.--->

the panic should be handled through a slice bounds check and surfaced to the caller

### Additional context

<!---Add any other context about the problem here.--->

The solution should be generic to catch / handle an error anytime a func is called prior to the instantiation of a job. 

Couple options:

1. Continue to enforce the Every/Cron methods be called first and catch the error

2. Allow methods to be called in any order, causing a new job to be created by whichever func is called first.

I like the 2nd approach better, and actually I think a bit of redesign here to think of this like a chain could be good allowing any order and the creation of a job to occur as implicitly in the first link of the chain.

--- a/example_test.go
+++ b/example_test.go
@@ -382,6 +382,7 @@ func ExampleScheduler_Month() {
 	_, _ = s.Every(1).Month(1).Do(task)
 	_, _ = s.Every(1).Months(1).Do(task)
 	_, _ = s.Every(1).Month(1, 2).Do(task)
+	_, _ = s.Month(1, 2).Every(1).Do(task)
 }
 
 func ExampleScheduler_Months() {

--- a/scheduler.go
+++ b/scheduler.go
@@ -981,7 +981,7 @@ func (s *Scheduler) Sunday() *Scheduler {
 }
 
 func (s *Scheduler) getCurrentJob() *Job {
-	if len(s.jobs) == 0 {
+	if len(s.Jobs()) == 0 {
 		job := &Job{}
 		s.setJobs(append(s.Jobs(), job))
 		s.isSetBehaviourBeforeJobCreated = true

--- a/scheduler.go
+++ b/scheduler.go
@@ -30,9 +30,10 @@ type Scheduler struct {
 
 	tags sync.Map // for storing tags when unique tags is set
 
-	tagsUnique      bool // defines whether tags should be unique
-	updateJob       bool // so the scheduler knows to create a new job or update the current
-	waitForInterval bool // defaults jobs to waiting for first interval to start
+	tagsUnique                     bool // defines whether tags should be unique
+	updateJob                      bool // so the scheduler knows to create a new job or update the current
+	waitForInterval                bool // defaults jobs to waiting for first interval to start
+	isSetBehaviourBeforeJobCreated bool // represents if set some feature before job created
 }
 
 // days in a week
@@ -439,22 +440,22 @@ func (s *Scheduler) NextRun() (*Job, time.Time) {
 // Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
 func (s *Scheduler) Every(interval interface{}) *Scheduler {
 	job := &Job{}
-	if s.updateJob {
+	if s.updateJob || s.isSetBehaviourBeforeJobCreated {
 		job = s.getCurrentJob()
 	}
 
 	switch interval := interval.(type) {
 	case int:
-		if !s.updateJob {
+		if !(s.updateJob || s.isSetBehaviourBeforeJobCreated) {
 			job = s.newJob(interval)
 		} else {
-			job.interval = interval
+			job = s.newJob(interval)
 		}
 		if interval <= 0 {
 			job.error = wrapOrError(job.error, ErrInvalidInterval)
 		}
 	case time.Duration:
-		if !s.updateJob {
+		if !(s.updateJob || s.isSetBehaviourBeforeJobCreated) {
 			job = s.newJob(0)
 		} else {
 			job.interval = 0
@@ -462,7 +463,7 @@ func (s *Scheduler) Every(interval interface{}) *Scheduler {
 		job.setDuration(interval)
 		job.setUnit(duration)
 	case string:
-		if !s.updateJob {
+		if !(s.updateJob || s.isSetBehaviourBeforeJobCreated) {
 			job = s.newJob(0)
 		} else {
 			job.interval = 0
@@ -474,7 +475,7 @@ func (s *Scheduler) Every(interval interface{}) *Scheduler {
 		job.setDuration(d)
 		job.setUnit(duration)
 	default:
-		if !s.updateJob {
+		if !(s.updateJob || s.isSetBehaviourBeforeJobCreated) {
 			job = s.newJob(0)
 		} else {
 			job.interval = 0
@@ -482,8 +483,11 @@ func (s *Scheduler) Every(interval interface{}) *Scheduler {
 		job.error = wrapOrError(job.error, ErrInvalidIntervalType)
 	}
 
-	if s.updateJob {
+	if s.updateJob || s.isSetBehaviourBeforeJobCreated {
 		s.setJobs(append(s.Jobs()[:len(s.Jobs())-1], job))
+		if s.isSetBehaviourBeforeJobCreated {
+			s.isSetBehaviourBeforeJobCreated = false
+		}
 	} else {
 		s.setJobs(append(s.Jobs(), job))
 	}
@@ -977,6 +981,11 @@ func (s *Scheduler) Sunday() *Scheduler {
 }
 
 func (s *Scheduler) getCurrentJob() *Job {
+	if len(s.jobs) == 0 {
+		job := &Job{}
+		s.setJobs(append(s.Jobs(), job))
+		s.isSetBehaviourBeforeJobCreated = true
+	}
 	return s.Jobs()[len(s.Jobs())-1]
 }
 
@@ -1032,7 +1041,7 @@ func (s *Scheduler) CronWithSeconds(cronExpression string) *Scheduler {
 
 func (s *Scheduler) cron(cronExpression string, withSeconds bool) *Scheduler {
 	job := s.newJob(0)
-	if s.updateJob {
+	if s.updateJob || s.isSetBehaviourBeforeJobCreated {
 		job = s.getCurrentJob()
 	}
 
@@ -1058,8 +1067,9 @@ func (s *Scheduler) cron(cronExpression string, withSeconds bool) *Scheduler {
 	job.setUnit(crontab)
 	job.startsImmediately = false
 
-	if s.updateJob {
+	if s.updateJob || s.isSetBehaviourBeforeJobCreated {
 		s.setJobs(append(s.Jobs()[:len(s.Jobs())-1], job))
+		s.isSetBehaviourBeforeJobCreated = false
 	} else {
 		s.setJobs(append(s.Jobs(), job))
 	}

--- a/scheduler.go
+++ b/scheduler.go
@@ -30,10 +30,10 @@ type Scheduler struct {
 
 	tags sync.Map // for storing tags when unique tags is set
 
-	tagsUnique                     bool // defines whether tags should be unique
-	updateJob                      bool // so the scheduler knows to create a new job or update the current
-	waitForInterval                bool // defaults jobs to waiting for first interval to start
-	isSetBehaviourBeforeJobCreated bool // represents if set some feature before job created
+	tagsUnique      bool // defines whether tags should be unique
+	updateJob       bool // so the scheduler knows to create a new job or update the current
+	waitForInterval bool // defaults jobs to waiting for first interval to start
+	jobCreated      bool // so the scheduler knows a job was created prior to calling Every or Cron
 }
 
 // days in a week
@@ -440,13 +440,13 @@ func (s *Scheduler) NextRun() (*Job, time.Time) {
 // Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
 func (s *Scheduler) Every(interval interface{}) *Scheduler {
 	job := &Job{}
-	if s.updateJob || s.isSetBehaviourBeforeJobCreated {
+	if s.updateJob || s.jobCreated {
 		job = s.getCurrentJob()
 	}
 
 	switch interval := interval.(type) {
 	case int:
-		if !(s.updateJob || s.isSetBehaviourBeforeJobCreated) {
+		if !(s.updateJob || s.jobCreated) {
 			job = s.newJob(interval)
 		} else {
 			job = s.newJob(interval)
@@ -455,7 +455,7 @@ func (s *Scheduler) Every(interval interface{}) *Scheduler {
 			job.error = wrapOrError(job.error, ErrInvalidInterval)
 		}
 	case time.Duration:
-		if !(s.updateJob || s.isSetBehaviourBeforeJobCreated) {
+		if !(s.updateJob || s.jobCreated) {
 			job = s.newJob(0)
 		} else {
 			job.interval = 0
@@ -463,7 +463,7 @@ func (s *Scheduler) Every(interval interface{}) *Scheduler {
 		job.setDuration(interval)
 		job.setUnit(duration)
 	case string:
-		if !(s.updateJob || s.isSetBehaviourBeforeJobCreated) {
+		if !(s.updateJob || s.jobCreated) {
 			job = s.newJob(0)
 		} else {
 			job.interval = 0
@@ -475,7 +475,7 @@ func (s *Scheduler) Every(interval interface{}) *Scheduler {
 		job.setDuration(d)
 		job.setUnit(duration)
 	default:
-		if !(s.updateJob || s.isSetBehaviourBeforeJobCreated) {
+		if !(s.updateJob || s.jobCreated) {
 			job = s.newJob(0)
 		} else {
 			job.interval = 0
@@ -483,10 +483,10 @@ func (s *Scheduler) Every(interval interface{}) *Scheduler {
 		job.error = wrapOrError(job.error, ErrInvalidIntervalType)
 	}
 
-	if s.updateJob || s.isSetBehaviourBeforeJobCreated {
+	if s.updateJob || s.jobCreated {
 		s.setJobs(append(s.Jobs()[:len(s.Jobs())-1], job))
-		if s.isSetBehaviourBeforeJobCreated {
-			s.isSetBehaviourBeforeJobCreated = false
+		if s.jobCreated {
+			s.jobCreated = false
 		}
 	} else {
 		s.setJobs(append(s.Jobs(), job))
@@ -983,7 +983,7 @@ func (s *Scheduler) Sunday() *Scheduler {
 func (s *Scheduler) getCurrentJob() *Job {
 	if len(s.Jobs()) == 0 {
 		s.setJobs([]*Job{{}})
-		s.isSetBehaviourBeforeJobCreated = true
+		s.jobCreated = true
 	}
 	return s.Jobs()[len(s.Jobs())-1]
 }
@@ -1040,7 +1040,7 @@ func (s *Scheduler) CronWithSeconds(cronExpression string) *Scheduler {
 
 func (s *Scheduler) cron(cronExpression string, withSeconds bool) *Scheduler {
 	job := s.newJob(0)
-	if s.updateJob || s.isSetBehaviourBeforeJobCreated {
+	if s.updateJob || s.jobCreated {
 		job = s.getCurrentJob()
 	}
 
@@ -1066,9 +1066,9 @@ func (s *Scheduler) cron(cronExpression string, withSeconds bool) *Scheduler {
 	job.setUnit(crontab)
 	job.startsImmediately = false
 
-	if s.updateJob || s.isSetBehaviourBeforeJobCreated {
+	if s.updateJob || s.jobCreated {
 		s.setJobs(append(s.Jobs()[:len(s.Jobs())-1], job))
-		s.isSetBehaviourBeforeJobCreated = false
+		s.jobCreated = false
 	} else {
 		s.setJobs(append(s.Jobs(), job))
 	}

--- a/scheduler.go
+++ b/scheduler.go
@@ -982,8 +982,7 @@ func (s *Scheduler) Sunday() *Scheduler {
 
 func (s *Scheduler) getCurrentJob() *Job {
 	if len(s.Jobs()) == 0 {
-		job := &Job{}
-		s.setJobs(append(s.Jobs(), job))
+		s.setJobs([]*Job{{}})
 		s.isSetBehaviourBeforeJobCreated = true
 	}
 	return s.Jobs()[len(s.Jobs())-1]

--- a/scheduler_test.go
+++ b/scheduler_test.go
@@ -1726,3 +1726,9 @@ func TestScheduler_CheckCalculateDaysOfMonth(t *testing.T) {
 		})
 	}
 }
+
+func TestScheduler_CheckSetBehaviourBeforeJobCreated(t *testing.T) {
+	s := NewScheduler(time.UTC)
+	s.Month(1, 2).Every(1).Do(func() {})
+
+}

